cmd/pulumi-test-language/tests.go

```diff
@@ -1641,6 +1642,74 @@ var languageTests = map[string]languageTest{
 			},
 		},
 	},
+	"l2-resource-parent-inheritance": {
+		providers: []plugin.Provider{&providers.SimpleProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					requireStackResource(l, err, changes)
+
+					// We expect the following resources:
+					//
+					// 0. The stack
+					//
+					// 1. The explicit simple provider, used to test provider inheritance.
+					// 2. The default simple provider, created in advance of the first resource that requires it.
+					//
+					// 3. A parent using the explicit provider.
+					// 4. A child of the parent using the explicit provider.
+					// 5. An orphan without a parent or explicit provider.
+					//
+					// 6. A parent with its protect flag set.
+					// 7. A child of the parent with its protect flag set.
+					// 8. An orphan without a parent or protect flag set.
+					require.Len(l, snap.Resources, 9, "expected 9 resources in snapshot")
+
+					explicitProvider := snap.Resources[1]
```

Comment:
Seeing some test failures https://github.com/pulumi/pulumi/actions/runs/12180944756/job/33976552858?pr=17936#step:40:123, I wonder if the order here is not deterministic?

Reply:
Yes, indeed. Hopefully pushed a fix.